examples/gno.land/p/demo/simpledao/propstore.gno

```diff
@@ -0,0 +1,173 @@
+package simpledao
+
+import (
+	"errors"
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/seqid"
+	"gno.land/p/gov/dao"
+)
+
+var (
+	errMissingProposal = errors.New("proposal is missing")
+	errMissingVote     = errors.New("member has not voted")
+)
+
+const (
+	// maxRequestProposals is the maximum number of
+	// paginated proposals that can be requested
+	maxRequestProposals = 10
+
+	// maxRequestMembers is the maximum number of
+	// paginated members that can be requested
+	maxRequestMembers = 50
+)
+
+// proposal is the internal simpledao proposal implementation
+type proposal struct {
+	author      std.Address // initiator of the proposal
+	description string      // description of the proposal
+
+	executor dao.Executor       // executor for the proposal
+	status   dao.ProposalStatus // status of the proposal
+
+	votes *votes // voting mechanism
+}
+
+func (p *proposal) GetAuthor() std.Address {
+	return p.author
+}
+
+func (p *proposal) GetDescription() string {
+	return p.description
+}
+
+func (p *proposal) GetStatus() dao.ProposalStatus {
+	return p.status
+}
+
+func (p *proposal) GetVotes() []dao.Vote {
+	var (
+		voters = p.votes.getVoters()
+		votes  = make([]dao.Vote, 0, voters.Size())
+	)
+
+	voters.Iterate("", "", func(key string, val interface{}) bool {
+		option := val.(dao.VoteOption)
+
+		vote := dao.Vote{
+			Address: std.Address(key),
+			Option:  option,
+		}
+
+		votes = append(votes, vote)
+
+		return false
+	})
+
+	return votes
+}
+
+func (p *proposal) GetVoteByMember(address std.Address) (dao.Vote, error) {
+	voters := p.votes.getVoters()
+
+	optionRaw, exists := voters.Get(address.String())
+	if !exists {
+		return dao.Vote{}, errMissingVote
+	}
+
+	vote := dao.Vote{
+		Address: address,
+		Option:  optionRaw.(dao.VoteOption),
+	}
+
+	return vote, nil
+}
+
+// PropStore implements the dao.PropStore abstraction
+type PropStore struct {
+	proposals *avl.Tree // seqid.ID -> proposal
+}
+
+// NewPropStore returns a new simpledao PropStore
+func NewPropStore() *PropStore {
+	return &PropStore{
+		proposals: avl.NewTree(),
+	}
+}
+
+// addProposal adds a new simpledao proposal to the store
+func (p *PropStore) addProposal(proposal *proposal) (uint64, error) {
+	// See what the next proposal number should be
+	nextID := uint64(p.proposals.Size())
+
+	// Save the proposal
+	p.proposals.Set(getProposalID(nextID), proposal)
+
+	return nextID, nil
+}
+
+func (p *PropStore) GetProposals(offset, count uint64) []dao.Proposal {
+	// Calculate the left and right bounds
+	if count < 1 {
+		return []dao.Proposal{}
+	}
```

Comment:
Comment is not right (maybe).

Reply:
Updated:

[30ac5ff](https://github.com/gnolang/gno/pull/2581/commits/30ac5ff8b77a74082eda6b4f882a16a2e249a330)